src/Core/src/Handlers/HybridWebView/HybridWebView.ts

```diff
@@ -0,0 +1,216 @@
+        };
+    };
+    webkit?: {
+        messageHandlers?: {
+            webwindowinterop?: {
+                postMessage: (message: any) => void;
+            };
+        };
+    };
+
+    // Declare the global object that we have added on Android.
+    hybridWebViewHost?: {
+        sendMessage: (message: string) => void;
+    };
+}
+
+/*
+ * The following interfaces define the shape of the messages that are sent between
+ * the web view and the .NET host application.
+ */
+interface JSInvokeMethodData {
+    MethodName: string;
+    ParamValues?: string[];
+}
+interface JSInvokeError {
+    Name?: string;
+    Message: string;
+    StackTrace?: string;
+}
+interface DotNetInvokeResult {
+    IsJson: boolean;
+    Result: any;
+}
+
+(() => {
+
+    /*
+     * Initialize the HybridWebView messaging system.
+     * This method is called once when the page is loaded.
+     */
+    function initHybridWebView() {
+        function dispatchHybridWebViewMessage(message: any) {
+            const event = new CustomEvent('HybridWebViewMessageReceived', { detail: { message: message } });
+            window.dispatchEvent(event);
+        }
+
+        if (window.chrome?.webview?.addEventListener) {
+            // Windows WebView2
+            window.chrome.webview.addEventListener('message', (arg: any) => {
+                dispatchHybridWebViewMessage(arg.data);
+            });
+        } else if (window.webkit?.messageHandlers?.webwindowinterop) {
+            // iOS and MacCatalyst WKWebView
+            // @ts-ignore - We are extending the global object here
+            window.external = {
+                receiveMessage: (message: any) => {
+                    dispatchHybridWebViewMessage(message);
+                },
+            };
+        } else {
+            // Android WebView
+            window.addEventListener('message', (arg: any) => {
+                dispatchHybridWebViewMessage(arg.data);
+            });
+        }
+    }
+
+    /*
+     * Send a message to the .NET host application.
+     * The message is sent as a string with the following format: `<type>|<message>`.
+     */
+    function sendMessageToDotNet(type: string, message: string) {
+        const messageToSend = type + '|' + message;
+
+        if (window.chrome && window.chrome.webview) {
+            // Windows WebView2
+            window.chrome.webview.postMessage(messageToSend);
+        } else if (window.webkit && window.webkit.messageHandlers && window.webkit.messageHandlers.webwindowinterop) {
+            // iOS and MacCatalyst WKWebView
+            window.webkit.messageHandlers.webwindowinterop.postMessage(messageToSend);
+        } else if (window.hybridWebViewHost) {
+            // Android WebView
+            window.hybridWebViewHost.sendMessage(messageToSend);
+        } else {
+            console.error('Unable to send messages to .NET because the host environment for the HybridWebView was unknown.');
+        }
```

Comment:
This logic to determine the platform is currently run on each call to `sendMessageToDotNet`. It's probably possible to run this logic once when the script initializes, and construct a `sendMessageToDotNet` function that directly calls the webview-specific function. I doubt the way it's currently written will create performance issues in practice though, so it's not critical to address this. Just a thought I had in passing :)